woocommerce/woocommerce-ios
# [Mobile Payments] Receipt Printing Controller is not shown

Closes: #7757 


## Description

Currently we do not show the `UIPrintInteractionController` when requested after collecting a payment, because we dismiss the underlying order flow `WooNavigationController` when showing it. As the system tries to present from there, we get this error:

`[Presentation] Attempt to present <UIPrintPanelNavigationController: 0x10c244c00> on <WooCommerce.WooNavigationController: 0x111808e00> (from <WooCommerce.WooNavigationController: 0x111808e00>) whose view is not in the window hierarchy.`

To solve it I make the receipt printing presentation _awaitable_, so we can close the order flow `WooNavigationController` **only** when the printing flow is finished, whatever its end result -success, error, cancel, etc.

## Testing instructions


1. Go to Menu.
2. Go to Payments.
3. Tap on Collect Payment.
4. Follow the flow and collect a payment with a card.
5. When the payment is collected successfully, tap on Print receipt.
6. The receipt printing controller should be presented -before it was not-.
7. Try different printing actions -email, save to notes, cancel, etc.-.
8. When done, it should dismiss the receipt printing flow, and then the order creation controller. (See attached video).

## Screenshots


https://user-images.githubusercontent.com/1864060/213214396-3b1e24e3-1434-494f-bf13-daccaf8404a4.MP4




---
